winrm/protocol.py

```diff
@@ -395,6 +395,30 @@ def cleanup_command(self, shell_id, command_id):
         # TODO change assert into user-friendly exception
         assert uuid.UUID(relates_to.replace('uuid:', '')) == message_id
 
+    def send_command_input(self, shell_id, command_id, stdin_input):
+        """
+        Send input to the given shell and command.
+        @param string shell_id: The shell id on the remote machine.
+         See #open_shell
+        @param string command_id: The command id on the remote machine.
+         See #run_command
+        @param string stdin_input: The input unicode string to be sent.
+        @return: None
+        """
+        stdin_input = text_type(stdin_input).encode("UTF-8")
+        req = {'env:Envelope': self._get_soap_header(
+            resource_uri='http://schemas.microsoft.com/wbem/wsman/1/windows/shell/cmd',  # NOQA
+            action='http://schemas.microsoft.com/wbem/wsman/1/windows/shell/Send',  # NOQA
+            shell_id=shell_id)}
+        stdin_envelope = req['env:Envelope'].setdefault('env:Body', {}).setdefault(
+            'rsp:Send', {}).setdefault('rsp:Stream', {})
+        stdin_envelope['@CommandId'] = command_id
+        stdin_envelope['@Name'] = 'stdin'
+        stdin_envelope['@xmlns:rsp'] = 'http://schemas.microsoft.com/wbem/wsman/1/windows/shell'
+        stdin_envelope['#text'] = base64.b64encode(stdin_input)
+        self.send_message(xmltodict.unparse(req))
+        #  @todo: result from send_message may need to be checked for errors, though I don't know what they would be.
```

Comment:
I don't think this todo is necessary, the actual response will contain a non-200 status code on an error which is raised in the transport section and the actual [SendResponse](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-wsmv/e7509be3-7794-4daf-b989-b1570a7f6b54) contains no information so there's nothing to check.

Reply:
If you are 100% sure.  I just didn't want anyone to be confused by any potential errors that are not handled.